File: java-rs/hello-world/src/main/java/HelloWorld.java
import static com.mongodb.client.model.Filters.eq;

import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import java.util.List;
import org.bson.Document;
import reactor.core.publisher.Mono;

public class HelloWorld {

    // A few sample product documents seeded by this app so you can run
    // it without loading an external dataset.
    private static final List<Document> SAMPLE_PRODUCTS = List.of(
        new Document("name", "Wireless Mouse")
            .append("category", "Electronics")
            .append("price", 24.99)
            .append("tags", List.of("wireless", "usb", "ergonomic")),
        new Document("name", "Standing Desk")
            .append("category", "Furniture")
            .append("price", 349.99)
            .append("tags", List.of("adjustable", "office")),
        new Document("name", "Noise-Cancelling Headphones")
            .append("category", "Electronics")
            .append("price", 199.99)
            .append("tags", List.of("bluetooth", "wireless", "over-ear"))
    );

    public static void main(String[] args) {
        String uri = System.getenv("MONGODB_URI");

        try (MongoClient client = MongoClients.create(uri)) {
            MongoDatabase database = client.getDatabase("get_started");
            MongoCollection<Document> products = database.getCollection("products");

            // Each reactive driver call returns a Publisher. Wrapping it in a
            // Reactor Mono and calling block() runs the operation and waits for
            // it to complete before moving on.

            // Seed the collection so the app has data to query. Clearing the
            // collection first keeps results consistent across repeated runs.
            Mono.from(products.deleteMany(new Document())).block();
            Mono.from(products.insertMany(SAMPLE_PRODUCTS)).block();

            Document product = Mono.from(products.find(eq("name", "Wireless Mouse")).first()).block();
            System.out.println(product.toJson());
        }
    }
}

File: kotlin-coroutine/hello-world/src/main/kotlin/HelloWorld.kt
import com.mongodb.client.model.Filters.eq
import com.mongodb.kotlin.client.coroutine.MongoClient
import kotlinx.coroutines.flow.firstOrNull
import kotlinx.coroutines.runBlocking
import org.bson.Document

// A few sample product documents seeded by this app so you can run
// it without loading an external dataset.
private val SAMPLE_PRODUCTS = listOf(
    Document("name", "Wireless Mouse")
        .append("category", "Electronics")
        .append("price", 24.99)
        .append("tags", listOf("wireless", "usb", "ergonomic")),
    Document("name", "Standing Desk")
        .append("category", "Furniture")
        .append("price", 349.99)
        .append("tags", listOf("adjustable", "office")),
    Document("name", "Noise-Cancelling Headphones")
        .append("category", "Electronics")
        .append("price", 199.99)
        .append("tags", listOf("bluetooth", "wireless", "over-ear"))
)

fun main() = runBlocking {
    val uri = System.getenv("MONGODB_URI")

    MongoClient.create(uri).use { client ->
        val database = client.getDatabase("get_started")
        val products = database.getCollection<Document>("products")

        // The coroutine driver exposes suspending functions, so these
        // calls run inside the runBlocking coroutine.

        // Seed the collection so the app has data to query. Clearing
        // the collection first keeps results consistent across
        // repeated runs.
        products.deleteMany(Document())
        products.insertMany(SAMPLE_PRODUCTS)

        val product = products.find(eq("name", "Wireless Mouse")).firstOrNull()
        println(product?.toJson())
    }
}
